### PR TITLE
Fix for #2434. String interpolation handling.

### LINKED ISSUE #2434: Syntax error (missing operator) in query expression
Recently I refactored some code and started to receive exceptions like `Syntax error (missing operator) in query expression '[t1].[Nam] [x].[Nam]'`. See `ORDER BY` with space between fields.

```SQL
SELECT
	[x].[Id],
	[t1].[Nam],
	[x].[Nam],
	[x].[Del]
FROM
...
ORDER BY
	[t1].[Nam] [x].[Nam]
```

This bug has been introduced because I changed details of expression methods
```c#
[ExpressionMethod(nameof(FullnameExpr), IsColumn = true)]
public string Fullname { get; set; }

private static Expression<Func<FtqWorkstationGroupDTO, string>> FullnameExpr() =>
```
from string concatenation
```c#
f => f.SectorPart.Name + " " + f.Name;
```
to string interpolation
```c#
f => $"{f.SectorPart.Name} {f.Name}";
```
With concatenation query is ok
```SQL
SELECT
	[x].[Id],
	[t1].[Nam],
	[x].[Nam],
	[x].[Del]
FROM
...
ORDER BY
	[t1].[Nam] + ' ' + [x].[Nam]
```
In my query I have ordering by `Fullname` like so:
```c#
return db.FtqWorkstationGroups
    .Where(x => x.SectorId == sectorId)
    .OrderBy(x => x.Fullname);
```
Is it some kind of limitation or a bug?

### Environment details
linq2db version: 3.0.1
Database Server: MS Access via OleDb

### PR DESCRIPTION
**Pull request by MaceWindu**
Fixes #2434

### CODE REVIEW

File: Tests/Linq/Linq/WhereTests.cs [hunk 1]
```diff
@@ -1570,6 +1570,22 @@ public void OptionalObjectInCondition([DataSources(false)] string context)
...
 
+		[Test]
+		public void SrtingInterpolationTests([DataSources(false)] string context)
+		{
+			using (var db = GetDataContext(context))
+			{
+				var cnt = db.Person.Count(p => p.LastName + ", " + p.FirstName == $"{p.LastName}, {p.FirstName}"
+				                               && "<" + p.LastName + ", " + p.FirstName + ">" ==
+				                               $"<{p.LastName}, {p.FirstName}>"
+				                               && "<" + p.LastName + p.FirstName + ">" == $"<{p.LastName}{p.FirstName}>"
```
Comment (MaceWindu) [inline Tests/Linq/Linq/WhereTests.cs:1581]: maybe add case with escaped {}? like `{{test}}`

Comment (sdanyliv) [reply Tests/Linq/Linq/WhereTests.cs:1581]: Good point, will add tests.

**Pull request merged**